### DISCUSSION TITLE
How do you block a specific domain using NXDOMAIN?

### DISCUSSION THREAD

**Discussion by devipasigner** [TechnitiumSoftware/DnsServer#537] [General]
Need this to block use-application-dns.net properly otherwise FireFox will still query DOH even with a null ip response

**Comment by ShreyasZare** [comment]
Thanks for asking. You can install the NXDOMAIN DNS app from the Apps section. The app by default blocks the `use-application-dns.net` domain. You can edit the app config to add more domain names.

**Reply by devipasigner** [reply]
Thank you. I tried this and ran `dig` but I get a noerror response with 0.0.0.0. I think the app may be broken?

Heres my config for the app:

```
{
  "enableBlocking": true,
  "allowTxtBlockingReport": true,
  "blocked": [
    "use-application-dns.net",
    "mask.icloud.com",
    "mask-h2.icloud.com"
  ]
}
```

**Reply by ShreyasZare** [reply]
Thanks for the details. Try `dig` again for TXT type and see what response you see. The TXT response should explain which app is blocking that domain name.

**Reply by devipasigner** [reply]
> Thanks for the details. Try `dig` again for TXT type and see what response you see. The TXT response should explain which app is blocking that domain name.

Seems like the advanced blocking app is blocking it. But shouldn’t the NXDOMAIN app be prioritized?

**Reply by ShreyasZare** [reply]
Currently the DNS apps design does not have any option to prioritize one app over another. They are chained in the sequence they are loaded.

**Reply by devipasigner** [reply]
> Currently the DNS apps design does not have any option to prioritize one app over another. They are chained in the sequence they are loaded.

So there is currently no way to get around this issue?

**Reply by ShreyasZare** [reply]
You can configure the Advanced Blocking app by adding the domain in its allowed list or if you have blocked it manually then unblock it. This way it wont block it and the call will get passed on to the next app to process it.
